beacon_node/beacon_chain/src/blob_verification.rs

```diff
@@ -254,27 +402,52 @@ pub fn validate_blob_sidecar_for_gossip<T: BeaconChainTypes>(
         });
     }
 
+    // Verify the inclusion proof in the sidecar
+    let _timer = metrics::start_timer(&metrics::BLOB_SIDECAR_INCLUSION_PROOF_VERIFICATION);
+    if !blob_sidecar
+        .verify_blob_sidecar_inclusion_proof()
+        .map_err(GossipBlobError::InclusionProof)?
+    {
+        return Err(GossipBlobError::InvalidInclusionProof);
+    }
+    drop(_timer);
+
+    let fork_choice = chain.canonical_head.fork_choice_read_lock();
+
     // We have already verified that the blob is past finalization, so we can
     // just check fork choice for the block's parent.
-    let Some(parent_block) = chain
-        .canonical_head
-        .fork_choice_read_lock()
-        .get_block(&block_parent_root)
-    else {
-        return Err(GossipBlobError::BlobParentUnknown(
-            signed_blob_sidecar.message,
-        ));
+    let Some(parent_block) = fork_choice.get_block(&block_parent_root) else {
+        return Err(GossipBlobError::BlobParentUnknown(blob_sidecar));
     };
 
+    // If fork choice does *not* consider the parent to be a descendant of the finalized block,
+    // then there are two more cases:
+    //
+    // 1. We have the parent stored in our database. Because fork-choice has confirmed the
+    //    parent is *not* in our post-finalization DAG, all other blocks must be either
+    //    pre-finalization or conflicting with finalization.
+    // 2. The parent is unknown to us, we probably want to download it since it might actually
+    //    descend from the finalized root.
+    if !fork_choice.is_finalized_checkpoint_or_descendant(block_parent_root) {
```

Comment:
I'm not sure how trivial the db check is, but we may not want to hold the fork choice lock while checking the block against the db?

Reply:
I was sus about this as well. However, we do something very similar in block verification
https://github.com/sigp/lighthouse/blob/44aaf13ff019e6c664f9244babee0fe39ea13149/beacon_node/beacon_chain/src/block_verification.rs#L818-L831

I would assume that checking existence of a block in the hot db is quite fast but not super sure. cc @michaelsproul

Reply:
And yeah, checking the block exists is _quite_ fast, but depends a bit on the disk. Should be 5ms on most good machines, which could be quite slow compared to the time for blob verif?